src/elements/content-sidebar/versions/__tests__/VersionsItem.test.js

```diff
@@ -238,5 +239,208 @@ describe('elements/content-sidebar/versions/VersionsItem', () => {
             expect(wrapper.exists(VersionsItemRetention)).toBe(true);
             expect(wrapper.find(VersionsItemActions).prop('isRetained')).toBe(true);
         });
+
+        test.each`
+            permissions
+            ${{ can_delete: true, can_download: true, can_preview: true, can_upload: true }}
+            ${{ can_delete: true, can_download: true, can_preview: true, can_upload: false }}
+            ${{ can_delete: true, can_download: true, can_preview: false, can_upload: true }}
+            ${{ can_delete: true, can_download: false, can_preview: true, can_upload: true }}
+            ${{ can_delete: false, can_download: true, can_preview: true, can_upload: true }}
+            ${{ can_delete: true, can_download: true, can_preview: false, can_upload: false }}
+            ${{ can_delete: true, can_download: false, can_preview: true, can_upload: false }}
+            ${{ can_delete: false, can_download: true, can_preview: true, can_upload: false }}
+            ${{ can_delete: false, can_download: true, can_preview: false, can_upload: true }}
+            ${{ can_delete: false, can_download: false, can_preview: true, can_upload: true }}
+            ${{ can_delete: false, can_download: true, can_preview: false, can_upload: false }}
+            ${{ can_delete: false, can_download: false, can_preview: true, can_upload: false }}
+        `(
+            'should disable the correct menu items based on permissions when unchangeableVersions feature is enabled',
+            ({ permissions }) => {
+                const unchangeableVersionsFeature = {
+                    versionsItem: {
+                        unchangeableVersions: {
+                            enabled: true,
+                        },
+                    },
+                };
+                const wrapper = getWrapper({
+                    version: getVersion({ permissions }),
+                    features: unchangeableVersionsFeature,
+                });
+                const actions = wrapper.find(VersionsItemActions);
+                const button = wrapper.find(VersionsItemButton);
+
+                expect(button.prop('isDisabled')).toBe(!permissions.can_preview);
+                expect(actions.prop('showDelete')).toBe(false);
+                expect(actions.prop('showDownload')).toBe(permissions.can_download);
+                expect(actions.prop('showPromote')).toBe(false);
+                expect(actions.prop('showPreview')).toBe(permissions.can_preview);
+                expect(wrapper.find(ReadableTime)).toBeTruthy();
```

Comment:
It'll always be false here, regardless of feature, since it needs action to be 'delete'. If I set action to delete it'll be testable, but same will happen to delete. I decided to test it with action tests below.

Reply:
I'll look if it can be done better after the weekend